### PR TITLE
Draft Pull Request for Author Toolkit

### PR DESCRIPTION
**Pull request by jbrock1999h**
This pull request adds codespace container files that will create the platform that authors use to easily edit their textbooks. The end goal is to have a button within the published textbook that links directly back to this codespace, allowing the textbook to be quickly and easily edited. So far, we have a codespace set up, but we do not yet have it as user friendly as we want it to be because it requires more git CLI manipulation than we want to get started.

This is an assignment for Josh Harvey and Daniel Short.

### CODE REVIEW

File: .devcontainer/devcontainer-20230420-004357.json
Comment (bjones1) [inline .devcontainer/devcontainer-20230420-004357.json:1]: Remove this file from the commit -- it was auto-edited, so we no longer care about this stuff.

File: .gitignore-20230420-004357
Comment (bjones1) [inline .gitignore-20230420-004357:1]: Remove this file also.

File: project-20230420-004357.ptx
Comment (bjones1) [inline project-20230420-004357.ptx:1]: Remove...

**Review by bjones1 — CHANGES_REQUESTED** [review]

**Review by jbrock1999h — COMMENTED** [review]
The requested files have been removed from the pull request

File: .devcontainer/postStartCommand.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# Reset to origin main (This is for if the codespace has not caught up to changes made previously)
+#git pull origin main
+#git reset --hard origin/main
+
+# Build pretext project
+#pretext build
+#pretext generate
+pretext view web &
+# The command pretext view web starts a local server which creates a process that does not end until the user manually ends it. This means that any command that comes after pretext view web will not run unless the local server is manually closed.
+xdg-open http://localhost:8000-.preview.app.github.dev
```
Comment (bjones1) [inline .devcontainer/postStartCommand.sh:14]: I don't understand how this command (open a specific URL) relates to the previous comment (which says this command won't run?).

File: .devcontainer/postCreateCommand.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
+# Create a new PreTeXt project.
+pretext new -d .
+
+# For the Digital System Design book...
```
Comment (bjones1) [inline .devcontainer/postCreateCommand.sh:6]: You can remove all this.

File: .devcontainer/docker-build.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+apt update
+apt install -y --no-install-recommends python3-pip texlive texlive-latex-extra texlive-fonts-extra texlive-xetex texlive-science texlive-music sagemath ghostscript pdf2svg
+
+# Upgrade pip and install required Python tools.
+python3 -m pip install --upgrade pip
+python3 -m pip install --upgrade CodeChat-Server runestone pretext
+
+apt install -y hub
+
+#apt install -y xdg-utils
```
Comment (bjones1) [inline .devcontainer/docker-build.sh:16]: You seem to run this later, so uncomment it? (But I don't understand why running it is necessary...)

File: .devcontainer/postStartCommand.sh [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# Add new remote repositories for origin (forked) and upstream (original)
+#git remote add upstream https://github.com/twjudson/aata
+#git remote add origin https://github.com/jbrock1999h/aata
+
+# Reset to origin main (This is for if the codespace has not caught up to changes made previously)
+#git pull origin main
+#git reset --hard origin/main
+
+# Build pretext project
+#pretext build
```
Comment (bjones1) [inline .devcontainer/postStartCommand.sh:10]: The pretext generate and pretext build should happen in the Docker container build, not here.

**Review by bjones1 — COMMENTED** [review]
Here are fixes/updates for your next PR; this is fine for the draft PR.

**Pull request merged**